src/main/java/tool/designpatterns/verifiers/multiclassverifiers/proxy/ProxyFeedbackUtils.java

```diff
@@ -0,0 +1,373 @@
+                    used = true;
+                    break;
+                }
+            }
+
+            if (!used) {
+                unused.add(classOrI);
+            }
+        }
+
+        return unused;
+    }
+
+    /**
+     * Returns the partial patterns that are not in any complete ProxyPatternGroup.
+     *
+     * @param partial  all the partial groups for either proxy or subject.
+     * @param complete the complete groups.
+     * @param pattern  the pattern (either subject or proxy) that are in the partial list.
+     *
+     * @return a new list containing the partial patterns that were not in any of the complete
+     *     groups.
+     */
+    private static List<PartialProxyImplementation> getCorrectPartial(
+        List<PartialProxyImplementation> partial, List<ProxyPatternGroup> complete,
+        Pattern pattern) {
+        List<PartialProxyImplementation> newPartial = new ArrayList<>();
+
+        for (PartialProxyImplementation partialGroup : partial) {
+            boolean isUsed = false;
+            for (ProxyPatternGroup completeGroup : complete) {
+                if (!isSameClassOrInterfaceDeclaration(partialGroup.getInterfaceOrAClass(),
+                    completeGroup.getInterfaceOrAClass())) {
+                    // The interface is not used.
+                    continue;
+                }
+
+                if (isSameClassOrInterfaceDeclaration(partialGroup.getInterfaceImplementor(),
+                    completeGroup.getFromPattern(pattern))) {
+                    // the pattern is also used.
+                    isUsed = true;
+                }
+            }
+
+            if (!isUsed) {
+                newPartial.add(partialGroup);
+            }
+        }
+
+        return newPartial;
+    }
+
+    /**
+     * Splits the given list in valid and invalid pattern groups.
+     *
+     * @param patternGroups the pattern groups to split.
+     *
+     * @return a tuple containing two new lists, the first containing only the valid
+     *     ProxyPatternGroups in the original list and the second containing only the invalid ones.
+     */
+    private static Tuple2<List<ProxyPatternGroup>, List<ProxyPatternGroup>> splitToValidInvalid(
+        List<ProxyPatternGroup> patternGroups) {
+        List<ProxyPatternGroup> invalid = new ArrayList<>();
+        List<ProxyPatternGroup> valid = new ArrayList<>();
+
+        for (ProxyPatternGroup group : patternGroups) {
+            if (group.isValid()) {
+                valid.add(group);
+            } else {
+                invalid.add(group);
+            }
+        }
+        return new Tuple2<>(valid, invalid);
+    }
+
+    /**
+     * Returns the feedbacks from the invalid ProxyPatternGroups.
+     *
+     * @param invalid the invalid ProxyPatternGroups.
+     * @param valid   the valid ProxyPatternGroupos.
+     *
+     * @return the feedbacks from the given ProxyPatternGroups.
+     */
+    private static List<Feedback> getInvalidFeedback(
```

Comment:
could have better name like "getInvalidGroupFeedbacks"

Reply:
Solved in 6fe6ab6